app/main.py

```diff
@@ -1,2 +1,51 @@
+from typing import Any
+
+
 class Dictionary:
     pass
+
+    def __init__(self) -> None:
+        self.size = 0
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.threshold = int(self.capacity * 2 / 3)
+        self.size = 0
+        self.hash_table = [[] for _ in range(self.capacity)]
+        for cell in old_hash_table:
+            if cell:
+                self.__setitem__(cell[0], cell[1])
+
+    def __setitem__(self, key: int, value: Any) -> None:
+        if self.size == self.threshold:
+            self.resize()
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, value, hash_]
+                self.size += 1
+                break
+
+            if self.hash_table[index][0] == key:
+                self.hash_table[index][1] = value
+                break
+
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key: int) -> Any:
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+        while self.hash_table[index]:
+            if self.hash_table[index][2] == hash_ and \
```

Comment:
It’s better to break line before `and` https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator